src/Client.lua

```diff
@@ -74,6 +74,21 @@ function Client:CreateWindow()
   return window
 end
 
+function Client:ConfirmShowRequest(sender)
+  -- Check whether the client (reciever) wants to recieve show requests from non officers
+  if NotaLoot:GetPref("ShowClientOfficers", true) then
+    if IsInGuild() then
+      local senderRank = select(1, NotaLoot:GetGuildRank(sender))
+      if senderRank and C_GuildInfo.GuildControlGetRankFlags(senderRank)[4] then
```

Comment:
Since the "is sender an officer" is now done in 2 places, I'm going to centralize the check in a separate function.